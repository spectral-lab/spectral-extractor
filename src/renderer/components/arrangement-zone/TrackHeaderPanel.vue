<template>
  <v-card
    :color="bgColor"
    class="track-header-panel"
    flat
  >
    <v-card-title primary-title>
      <h3>
        {{ idx + 1 }} {{ track.name }}
      </h3>
    </v-card-title>
    <v-card-actions>
      <v-layout row>
        <v-spacer />
        <icon-btn-with-tip
          @click="handleClick"
          :color="bgColor"
          icon="fa-cog"
          tip="Tack setting"
        />
      </v-layout>
    </v-card-actions>
  </v-card>
</template>

<script>
// @flow
import IconBtnWithTip from '../misc/IconBtnWithTip';
import Color from 'color';
import Vue from 'vue';
import { TRACK } from '../../../constants/model-types';
import { INSPECT } from '../../../constants/dialog-types';
import { dialogEventHub } from '../../modules';

export default Vue.extend({
  components: {
    IconBtnWithTip
  },
  props: {
    idx: {
      type: Number,
      default: 1
    },
    track: {
      type: Object,
      default: () => ({})
    }
  },
  computed: {
    bgColor () {
      return Color(this.track.color).darken(0.5).rgb().string();
    }
  },
  methods: {
    handleClick () {
      dialogEventHub.emit(null, {
        type: INSPECT,
        context: TRACK,
        id: this.track.id
      });
    }
  }
});
</script>

<style scoped>

</style>
