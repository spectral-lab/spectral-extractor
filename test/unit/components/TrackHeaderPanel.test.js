import { shallowMount } from '@vue/test-utils';
import TrackHeaderPanel from '../../../src/renderer/components/arrangement-zone/TrackHeaderPanel';
import Vue from 'vue';
import vuetify from 'vuetify';
import mockTrack from '../../data/json/mockTrack.json';

Vue.use(vuetify);

describe('Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(TrackHeaderPanel, {
      propsData: {
        idx: 2,
        track: mockTrack
      }
    });
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
