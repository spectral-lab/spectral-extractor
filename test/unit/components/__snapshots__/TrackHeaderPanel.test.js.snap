// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  class="track-header-panel"
>
  <div
    class="color-label"
    style="background-color: rgb(45, 43, 67);"
  />
   
  <div
    class="panel-content"
    style="background-color: rgb(35, 36, 39);"
  >
    <v-card-title-stub>
      <h3
        style="color: white;"
      >
        
        3. Track
      
      </h3>
    </v-card-title-stub>
     
    <v-card-actions-stub>
      <v-layout-stub
        row=""
        tag="div"
      >
        <v-spacer-stub />
         
        <icon-btn-with-tip-stub
          color="rgb(35,36,39)"
          icon="fa-cog"
          tip="Tack setting"
        />
      </v-layout-stub>
    </v-card-actions-stub>
  </div>
</div>
`;
